sql/sp_rcontext.h

```diff
@@ -144,7 +144,7 @@ class sp_rcontext : public Sql_alloc
       memcpy(sql_state, _sql_state, SQLSTATE_LENGTH);
       sql_state[SQLSTATE_LENGTH]= '\0';
 
-      strncpy(message, _message, MYSQL_ERRMSG_SIZE - 1);
+      strncpy(message, _message, MYSQL_ERRMSG_SIZE);
```

Comment:
What was the warning here? I am curious only because this type of warnings we usually fix with decreasing the number of bytes being copied.

Reply:
```
In file included from /usr/include/string.h:494,
                 from /data/mysql-server/PS-5761-5.6/include/m_string.h:34,
                 from /data/mysql-server/PS-5761-5.6/sql/sql_plugin.h:34,
                 from /data/mysql-server/PS-5761-5.6/sql/structs.h:23,
                 from /data/mysql-server/PS-5761-5.6/sql/unireg.h:170,
                 from /data/mysql-server/PS-5761-5.6/sql/sp_rcontext.cc:17:
In function ‘char* strncpy(char*, const char*, size_t)’,
    inlined from ‘sp_rcontext::Sql_condition_info::Sql_condition_info(uint, const char*, Sql_condition::enum_warning_level, const char*)’ at /data/mysql-server/PS-5761-5.6/sql/sp_rcontext.h:147:14,
    inlined from ‘sp_rcontext::Handler_call_frame::Handler_call_frame(const sp_handler*, uint, const char*, Sql_condition::enum_warning_level, const char*, uint)’ at /data/mysql-server/PS-5761-5.6/sql/sp_rcontext.h:185:31,
    inlined from ‘bool sp_rcontext::handle_sql_condition(THD*, uint*, const sp_instr*)’ at /data/mysql-server/PS-5761-5.6/sql/sp_rcontext.cc:400:54:
/usr/include/x86_64-linux-gnu/bits/string_fortified.h:106:34: error: ‘char* __builtin_strncpy(char*, const char*, long unsigned int)’ output may be truncated copying 511 bytes from a string of length 511 [-Werror=stringop-truncation]
```

Reply:
The change is safe as the next line is `message[MYSQL_ERRMSG_SIZE - 1]= '\0';`